src/platform/plugins/shared/data_views/common/data_views/data_view.ts

```diff
@@ -340,6 +340,7 @@ export class DataView extends AbstractDataView implements DataViewBase {
     if (existingField && existingField.isMapped) {
       // mapped field, remove runtimeField def
       existingField.runtimeField = undefined;
+      this.fields.clearSpecCache();
```

Comment:
Built on top of this PR: https://github.com/elastic/kibana/pull/224767 to test performance